flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackendBuilder.java

```diff
@@ -497,21 +507,25 @@ private RocksDBRestoreOperation getRocksDBRestoreOperation(
             long lastCompletedCheckpointId) {
         RocksDBSnapshotStrategyBase<K, ?> checkpointSnapshotStrategy;
         if (enableIncrementalCheckpointing) {
+            RocksDBStateUploader stateUploader =
+                    injectRocksDBStateUploader == null
+                            ? new RocksDBStateUploader(numberOfTransferingThreads)
+                            : injectRocksDBStateUploader;
             checkpointSnapshotStrategy =
                     new RocksIncrementalSnapshotStrategy<>(
-                            db,
-                            rocksDBResourceGuard,
-                            keySerializerProvider.currentSchemaSerializer(),
-                            kvStateInformation,
-                            keyGroupRange,
-                            keyGroupPrefixBytes,
-                            localRecoveryConfig,
-                            cancelStreamRegistry,
-                            instanceBasePath,
-                            backendUID,
-                            materializedSstFiles,
-                            lastCompletedCheckpointId,
-                            numberOfTransferingThreads);
+                                    db,
+                                    rocksDBResourceGuard,
+                                    keySerializerProvider.currentSchemaSerializer(),
+                                    kvStateInformation,
+                                    keyGroupRange,
+                                    keyGroupPrefixBytes,
+                                    localRecoveryConfig,
+                                    cancelStreamRegistry,
+                                    instanceBasePath,
+                                    backendUID,
+                                    materializedSstFiles,
+                                    lastCompletedCheckpointId)
+                            .setRocksDBStateUploader(stateUploader);
```

Comment:
I think there is no need to introduce a setter in `RocksIncrementalSnapshotStrategy`. 
`stateUploader` can be passed to its constructor and be a final field there (in fact, I think one of the benefits of having a builder is immutability of the object it builds).
WDYT?